web/api/v1/api.go

```diff
@@ -476,6 +480,11 @@ func (api *API) query(r *http.Request) (result apiFuncResult) {
 		return apiFuncResult{nil, returnAPIError(res.Err), res.Warnings, qry.Close}
 	}
 
+	var warnings annotations.Annotations
+	if limit > 0 {
+		res = truncateResults(res, limit)
+		warnings = warnings.Add(errors.New("results truncated due to limit"))
```

Comment:
You should only add the warning if the results were actually truncated.

`truncateResults` could return an additional `bool` telling the caller if a truncation has happened.

Reply:
done.